content/cpp/concepts/forward-list/forward-list.md

```diff
@@ -0,0 +1,248 @@
+
+std::cout << "Calling list.clear()" << std::endl;
+list.clear();
+
+std::cout << "Elements of list: ";
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+
+std::cout << "list.empty(): ";
+std::cout << list.empty() << std::endl;
+```
+
+```shell
+Elements of list: 1 2 3 4 5
+list.empty(): 0
+Calling list.clear()
+Elements of list:
+list.empty(): 1
+```
+
+`.front()` returns the value of the first element.
+
+```cpp
+std::forward_list<int> list = {1,2,3,4,5};
+
+std::cout << "Output: " << list.front() << endl;
+
+```
+
+```shell
+Output: 1
+```
+
+`.push_front()` inserts the provided value at the front of the list.
+
+```cpp
+std::forward_list<int> list = {1,2,3,4,5};
+
+list.push_front(0);
+
+std::cout << "Output:" << std::endl;
+
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+```
+
+```shell
+Output:
+0 1 2 3 4 5
+```
+
+`.pop_front()` removes the first element from the list.
+
+```cpp
+std::forward_list<int> list = {1,2,3,4,5};
+
+list.pop_front();
+
+std::cout << "Output:" << std::endl;
+
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+```
+
+```shell
+Output:
+2 3 4 5
+```
+
+`.insert_after()` inserts an element after an iterator pointer.
+
+`.erase_after()` removes the element after an iterator pointer.
+
+```cpp
+std::forward_list<int> list = {2,5,8,10,12};
+
+std::cout << "Elements of list: ";
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+
+auto iter = list.begin(); //Points to the first element (2)
+iter++; //Now points to the second element (5)
+
+std::cout << "Calling list.insert_after(iter, 6) while iter points to 5" << std::endl;
+list.insert_after(iter, 6);
+
+std::cout << "Elements of list: ";
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+
+iter++; //Now points to the third element (6)
+
+std::cout << "Calling list.erase_after(iter) while iter points to 6" << std::endl;
+list.erase_after(iter);
+
+std::cout << "Elements of list: ";
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+```
+
+```shell
+Elements of list: 2 5 8 10 12
+Calling list.insert_after(iter, 6) while iter points to 5
+Elements of list: 2 5 6 8 10 12
+Calling list.erase_after(iter) while iter points to 6
+Elements of list: 2 5 6 10 12
+```
+
+`.assign()` replaces the current list with a new list.
+
+```cpp
+std::forward_list<int> list = {1,2,3,4,5};
+
+std::cout << "Elements: ";
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+
+std::cout << "Calling list.assign({6, 7})" << std::endl;
+list.assign({6, 7});
+
+
+std::cout << "Elements: ";
+for (const auto& element : list) {
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+```
+
+```shell
+Elements: 1 2 3 4 5
+Calling list.assign({6, 7})
+Elements: 6 7
+```
```

Comment:
Please remove all this and just a Codebyte Example section that provides an example demonstrating how to use forward_list